api/types/authentication.go

```diff
@@ -775,14 +775,11 @@ func (c *AuthPreferenceV2) CheckAndSetDefaults() error {
 		return trace.BadParameter("missing required Webauthn configuration for headless=true")
 	}
 
-	// Prevent accidental local lockout by disabling local second factor methods, (likely leaving only sso enabled).
+	// Prevent local lockout by disabling local second factor methods.
+	hasSSO := c.IsSecondFactorSSOAllowed()
 	hasTOTP := c.IsSecondFactorTOTPAllowed()
-	if c.GetAllowLocalAuth() && !hasTOTP && !hasWebauthn {
-		switch c.Spec.SecondFactor {
-		case constants.SecondFactorOptional, constants.SecondFactorOff:
-		default:
-			return trace.BadParameter("missing a local second factor method for local users (otp, webauthn), either add a local second factor method or disable local auth")
-		}
+	if c.GetAllowLocalAuth() && hasSSO && !hasTOTP && !hasWebauthn {
```

Comment:
Why does `hasSSO` matter here? It looks like the test doesn't really care whether it's true of false.

Reply:
The test does set `SecondFactors=[sso]`, but what we really care about is `if second factor is enforced && local second factors are disabled`. I'll add a `IsLocalMFAEnabled` helper to clarify things.